### DISCUSSION TITLE
Can we get annotation framework generated files to include the opening comment:    // <auto-generated/>

### DISCUSSION THREAD

**Discussion by lorenh** [aws/aws-lambda-dotnet#1733] [General]
We are using StyleCop.Analyzers and it's including the *_Generated.g.cs, and Program.g.cs files generated by the Amazon.Lambda.Annotations.SourceGenerator.Generator.  It would be nice if they included

// &lt;auto-generated/&gt;

at the top of the file like Microsoft generators do, so they automatically get skipped by the analyzers.  Is there any way to do this today, or would this be a change request.

**Comment by normj** [comment]
Is that documented somewhere that if we add `// <auto-generated/>` it will get skipped? If so it would be simple enough to add the source generator to put that in the file.

**Reply by lorenh** [reply]
That's a good question. I see it in all of the source generated by Microsoft code generators, and I see it in this sample:
   https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators 

Looking for "official" documentation on it though.

Oh, here's something:

https://stackoverflow.com/questions/63073472/roslyn-how-detect-code-files-that-are-auto-generated

https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs

**Comment by normj** [comment]
Thanks for the links. It does make sense to add the line to the generated code.

**Comment by 96malhar** [comment]
PR addressing this - https://github.com/aws/aws-lambda-dotnet/pull/1748

**Comment by ashishdhingra** [comment]
This should have been addressed via PR https://github.com/aws/aws-lambda-dotnet/pull/1748 and available in [Amazon.Lambda.Annotations v1.4.0](https://www.nuget.org/packages/Amazon.Lambda.Annotations/1.4.0) package.
